packages/override_api_endpoint/lib/src/override_api_endpoint.dart

```diff
@@ -0,0 +1,34 @@
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:uni_links/uni_links.dart';
+
+Future<Uri> overrideApiEndpoint(
```

Comment:
1. Please add dartdoc comment with documentation for this function.
2. Please make it a class that would receive the `SharedPreferences` and `Future<Uri?> Function()` (typedefed most likely) to constructor so that it's easily mockable and hence testable/debuggable. And make the function a method of this class

Reply:
@lukaszgarstecki I'd also add dartdoc comments with examples because if I was new to the topic, I'd actually wonder what I should pass as those segment/queryparameter etc. arguments

Reply:
@lukaszgarstecki @Albert221 I'm not sure if making it a class is necessary, to make it testable we can just pass sharedPrefs and that function as parameters with default values, so the usage of this function would stay the same as now

Reply:
Done